File: android/app/src/main/kotlin/io/dotnetnativeinterop/trust/TrustModels.kt
package io.dotnetnativeinterop.trust

import kotlinx.serialization.Serializable

/**
 * The `trust~posture` payload — the honest per-transport security posture the engine reports (camelCase,
 * mirroring the engine's `TrustPostureReport` / `TransportPosture` / `PqChannelParams` records). Honesty
 * is the whole point: HTTP reports as plaintext loopback, and the binary transport reports plaintext
 * until (and unless) a PQ handshake actually completes — at which point [binaryPqChannel] carries the
 * LIVE negotiated params, not a hardcoded label.
 */
@Serializable
public data class TransportPosture(
    val transport: String,   // "ffi" | "http" | "sqlcipher" | "binary"
    val inProcess: Boolean,
    val encrypted: Boolean,
    val wire: String,
    val detail: String,
)

/** Live params of an active framed-protobuf PQ channel (null in the report when plaintext / no channel). */
@Serializable
public data class PqChannelParams(
    val kem: String,
    val sig: String,
    val cipher: String,
    val kemPublicKeyBytes: Int,
    val ciphertextBytes: Int,
    val sharedSecretBytes: Int,
    val handshakeUs: Double,
)

/** Full trust posture: per-transport posture + the live PQ params when the binary channel is up. */
@Serializable
public data class TrustPostureReport(
    val transports: List<TransportPosture> = emptyList(),
    val binaryPqChannel: PqChannelParams? = null,
)

File: ios/Shared/Trust/TrustModels.swift
import Foundation

/// Codable mirror of the engine's `trust~posture` JSON (`TrustPosture.ReportJson`, camelCase). Honesty is
/// the whole point: HTTP reports as plaintext loopback, and the binary transport reports plaintext until —
/// and unless — a real ML-KEM/ML-DSA handshake completes, at which point `binaryPqChannel` carries the
/// LIVE negotiated params (published by the server, not a hardcoded label).

/// The security posture of one interop transport, told honestly.
struct TransportPosture: Codable, Identifiable, Sendable {
    let transport: String   // "ffi" | "http" | "sqlcipher" | "binary"
    let inProcess: Bool
    let encrypted: Bool
    let wire: String
    let detail: String

    var id: String { transport }
}

/// Live params of an active framed-protobuf PQ channel (null in the report when plaintext / no channel).
/// A Codable peer of the handshake-side `PqChannelParams` — same fields, but decoded from the server's
/// report rather than produced by the client handshake.
struct TrustPqChannelParams: Codable, Sendable {
    let kem: String
    let sig: String
    let cipher: String
    let kemPublicKeyBytes: Int
    let ciphertextBytes: Int
    let sharedSecretBytes: Int
    let handshakeUs: Double
}

/// Full trust posture: per-transport posture + the live PQ params when the binary channel is up.
struct TrustPostureReport: Codable, Sendable {
    let transports: [TransportPosture]
    let binaryPqChannel: TrustPqChannelParams?
}
